bpf/lib/fib.h

```diff
@@ -78,12 +78,6 @@ fib_do_redirect(struct __ctx_buff *ctx, const bool needs_l2_check,
 		const struct bpf_fib_lookup_padded *fib_params,
 		bool allow_neigh_map, __s8 *fib_ret, int *oif)
 {
-	/* sanity check, we only enter this function with these two fib lookup
```

Comment:
We have a couple plain calls to `fib_do_redirect`.

https://github.com/cilium/cilium/blob/28b76caffb997bc3182db9defbe71e1d07ddd745/bpf/lib/nodeport.h#L134

https://github.com/cilium/cilium/blob/28b76caffb997bc3182db9defbe71e1d07ddd745/bpf/lib/egress_gateway.h#L55

These do check the return from fib_lookup before calling `fib_do_redirect`. However, are we always going to catch this during PR if a naked call to `fib_do_redirect` is done without checking the error? This is why this catch was here, or else wrong use goes right into the direct and maybe a bit harder to catch.

Reply:
Calling `fib_do_redirect()` directly is "expert mode" - so yes, I'd expect users to know what they're doing here. In particular as all existing users are doing it correctly atm, and there's no risk of developers stumbling over a "bad" example.